### PR TITLE
 Added yaml configuration instructions

### PR DESCRIPTION
**Pull request by eminn**

### CODE REVIEW

File: src/docs/asciidoc/configuration.adoc [hunk 1]
```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
...
+
+NOTE: Use the `<import/>` element on top level of the XML hierarchy.
+
+NOTE: Use the `import` mapping on top level of the YAML hierarchy.
+
+Hazelcast Jet and Hazelcast IMDG share same syntax for declarative
+configuration composition. For more examples on importing resources from
+the classpath and file system or resources with variables in their 
+names, refer to the {hz-refman}#composing-declarative-configuration[relevant section] 
+of Hazelcast IMDG reference manual.
```
Comment (Serdaro) [inline src/docs/asciidoc/configuration.adoc]: in the Hazelcast IMDG Reference Manual.

File: src/docs/asciidoc/configuration.adoc [hunk 2]
```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
...
+    - development-metrics-config.yaml
+    - development-instance-config.yaml
+----
+
+NOTE: Use the `<import/>` element on top level of the XML hierarchy.
+
+NOTE: Use the `import` mapping on top level of the YAML hierarchy.
+
+Hazelcast Jet and Hazelcast IMDG share same syntax for declarative
+configuration composition. For more examples on importing resources from
```
Comment (Serdaro) [inline src/docs/asciidoc/configuration.adoc]: ... importing the resources ...

File: src/docs/asciidoc/configuration.adoc [hunk 3]
```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
...
+the classpath and file system or resources with variables in their
```
Comment (Serdaro) [inline src/docs/asciidoc/configuration.adoc]: ... file system, or the resources with ...

File: src/docs/asciidoc/configuration.adoc [hunk 4]
```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
...
+Hazelcast Jet and Hazelcast IMDG share same syntax for declarative
+configuration composition. For more examples on importing resources from
+the classpath and file system or resources with variables in their 
+names, refer to the {hz-refman}#composing-declarative-configuration[relevant section] 
+of Hazelcast IMDG reference manual.
+
+=== Using Variables
+In your Hazelcast Jet declarative configuration, you can use variables 
+to set the values of the elements. This is valid when you set a system 
+property programmatically or you use the command line interface. You can
```
Comment (Serdaro) [inline src/docs/asciidoc/configuration.adoc]: ... or use the command line ...

File: src/docs/asciidoc/configuration.adoc [hunk 5]
```diff
@@ -238,9 +270,200 @@ An example `hazelcast-jet.xml` is shown below :
...
+
+Hazelcast Jet supports variable replacers which are used to replace 
+custom strings during loading the configuration.
+
+Hazelcast Jet and Hazelcast IMDG share same syntax for variable
+replacers. You can configure your variable replacers in Hazelcast Jet 
+configuration via XML or YAML. Please see {hz-refman}#variable-replacers[Variable Replacers]
+section for more information and examples about variable replacers on 
+Hazelcast IMDG reference manual.
+
```
Comment (Serdaro) [inline src/docs/asciidoc/configuration.adoc]: My suggestion for the last sentence:

.... Please see the {hz-refman}#variable-replacers[Variable Replacers section]
in the Hazelcast IMDG Reference Manual for more information and examples on the
variable replacers.

**Comment by olukas** [comment]
Please add an information that in case when yaml file is used in `hazelcast.jet.config` then it has to have file extension `.yaml` or `.yml`. Otherwise it is not treated as yaml config file.

**Pull request merged**